src/algebra/discrete-log.md

```diff
@@ -6,10 +6,7 @@ The discrete logarithm is an integer $x$ solving the equation
 
 $$a^x \equiv b \pmod m,$$
 
-where $a$ and $m$ are relatively prime.
-**Note**, if they are not relatively prime, then the algorithm described below is incorrect, though it can be modified so that it can work.
-
-In this article, we describe the **Baby step - giant step** algorithm, proposed by Shanks in 1971, which has the time complexity $O(\sqrt{m} \log m)$. This algorithm is also known as **meet-in-the-middle** because it uses the technique of separating tasks in half.
+In this article, we describe the **Baby step - giant step** algorithm, proposed by Shanks in 1971, which has the time complexity $O(\sqrt{m})$. This is a **meet-in-the-middle** algorithm because it uses the technique of separating tasks in half.
```

Comment:
Can we add something like: "In this article, we describe the Baby step - giant step algorithm, **an algorithm to compute the discrete logarithm** proposed by Shanks in 1971, ..."

Reply:
Done.